pipeline-templates/build-test.yaml

```diff
@@ -0,0 +1,38 @@
+parameters:
+  pool: ''
+
+jobs:
+- job: ${{ parameters.pool.os }}_Build
+  pool:
+    ${{ if eq(parameters.oes, 'true') }}:
+        ${{ parameters.pool }}
+    ${{ else }}:
+        vmImage: ${{ parameters.pool.vmImage }}

```

Comment:
Oh, I see how to resolved the "you can't use the same templates" problem. That works just fine. I'd likely recommend using an actual parameter name instead of an acronym like `oes` which isn't very descriptive, and kinda looks like a typo for `os`.

Reply:
Changed to `useOneEngineeringPool`, good feedback.